### PR TITLE
Shwetank tt/config

### PR DESCRIPTION
**Pull request by shwetankTT**
Cleaning up Block config parameter for Convolution Api. Removed those variables which can be iterator through parallel Configs. Renaming per_core_weight_matrix_width_ntiles --> per_core_out_matrix_width_ntiles.

Tested with test_conv2d api test.

There are multiple commits in this PR. I will merge them into one commit at the time of merging into base.

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py [hunk 1]
```diff
@@ -890,40 +863,41 @@ def conv_input_interleaved_to_sharded(self, conv_input_on_device: ttl.tensor.Ten
...
-                    (int)(padded_input_channels / act_c_num_blocks),
+                    padded_input_channels,
                 ],  # act_block_w_datums may include reads of multiple pixels in window
-                ttl.tensor.TensorMemoryLayout.BLOCK_SHARDED,
-                ttl.tensor.ShardOrientation.COL_MAJOR,
+                ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED,
+                ttl.tensor.ShardOrientation.ROW_MAJOR,
             )
         else:
+            act_c_num_blocks = self.opt_conv_parall_conf_auto.grid_size.y
```
Comment (nsmithtt) [inline tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py:885]: IMO, this variable is clearer as just `grid_size_y` or something.  Or perhaps `grid_size_n` for matmul `mkn` dims.

**Review by nsmithtt — APPROVED** [review]

**Comment by nsmithtt** [comment]
LGTM if CI is happy :)

**Review by arakhmati — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]
Please make sure you run the CI
(https://github.com/tenstorrent-metal/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
on this branch and merge after it passes all the tests.

File: models/demos/resnet/tests/test_metal_resnet50.py [hunk 2]
```diff
@@ -118,7 +118,7 @@
 
 
 @skip_for_wormhole_b0("This test is not supported on WHB0, please use the TTNN version.")
-@pytest.mark.parametrize("batch_size", [1, 2, 8, 16, 20], ids=["batch_1", "batch_2", "batch_8", "batch_16", "batch_20"])
```
Comment (mywoodstock) [inline models/demos/resnet/tests/test_metal_resnet50.py:121]: Can you also add a NOTE saying that batch 1 and 2 are not yet supported with sharded data?

**Review by mywoodstock — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py [hunk 3]
```diff
@@ -366,9 +340,6 @@ class TTPyCompositeConv(TTPyOp):
         "per_core_weight_matrix_width",
         "act_block_h",
         "act_block_w",
-        "act_c_num_blocks",
-        "weight_block_w",
-        "out_block_h",
         "out_block_w",
```
Comment (tt-nshanker) [inline tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py:343]: do we need "out_block_w" or can it be derived?

Comment (tt-nshanker) [reply tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py:343]: Looks like "out_block_w" is unused and not set at all. We can probably delete it from config_keys in another check-in.

**Review by tt-nshanker — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py (hunk 3 above)
Comment (shwetankTT) [reply tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_composite_conv.py:343]: > Looks like "out_block_w" is unused and not set at all. We can probably delete it from config_keys in another check-in.

Sure. Noted.

File: tt_eager/tt_dnn/op_library/conv/multi_core_optimized_conv/optimized_conv_op.cpp [hunk 4]
```diff
@@ -234,6 +233,7 @@ operation::ProgramWithCallbacks multi_core_optimized_conv_(const Tensor& a, cons
     assert(weight_matrix_width_ntiles % weight_block_w_ntiles == 0);
     assert(act_matrix_height_ntiles % out_block_h_ntiles == 0);
 
+    uint32_t conv_act_c_blocks = 1;  // input is HS
```
[the remark this answers is not in the document]
Comment (mywoodstock) [reply tt_eager/tt_dnn/op_library/conv/multi_core_optimized_conv/optimized_conv_op.cpp:236]: No I don't think it does. Let's ditch it

**Pull request merged**